### PR TITLE
Deduplicate data block buffers

### PR DESCRIPTION
**Pull request by Shane32**
Small performance update to prevent duplication of data.

(Note: will create merge conflicts with #519 )

### CODE REVIEW

File: QRCoder/QRCodeGenerator.cs [hunk 1]
```diff
@@ -224,17 +224,17 @@ private static QRCodeData GenerateQrCode(BitArray bitArray, ECCLevel eccLevel, i
...
+            AddCodeWordBlocks(2, eccInfo.BlocksInGroup2, eccInfo.CodewordsInGroup2, offset, bitArray.Length - offset);
 
 
             //Calculate interleaved code word lengths
             int interleavedLength = 0;
             for (var i = 0; i < Math.Max(eccInfo.CodewordsInGroup1, eccInfo.CodewordsInGroup2); i++)
             {
                 foreach (var codeBlock in codeWordWithECC)
-                    if (codeBlock.CodeWords.Length > i)
+                    if (codeBlock.CodeWordsLength / 8 > i)
```
[the remark this answers is not in the document]
Comment (Shane32) [reply QRCoder/QRCodeGenerator.cs:237]: Done

**Comment by codebude** [comment]
Hm while trying to resolve the merge conflict, I broke the code. I'm not sure where I made the mistake. :-( @Shane32 can you help out?

**Comment by Shane32** [comment]
Sure just a minute.  I knew there'd be merge conflicts.

File: QRCoder/QRCodeGenerator.cs [hunk 2]
```diff
@@ -289,17 +291,14 @@ private static QRCodeData GenerateQrCode(BitArray bitArray, ECCLevel eccLevel, i
...
 
-            void AddCodeWordBlocks(int blockNum, int blocksInGroup, int codewordsInGroup, BitArray bitArray2, int offset2, int count, Polynom generatorPolynom)
+
+            void AddCodeWordBlocks(int blockNum, int blocksInGroup, int codewordsInGroup, int offset2, int count, Polynom generatorPolynom)
             {
                 var groupLength = codewordsInGroup * 8;
                 for (var i = 0; i < blocksInGroup; i++)
                 {
-                    var bitBlockList = BinaryStringToBitBlockByteList(bitArray2, offset2, groupLength);
                     var eccWordList = CalculateECCWords(bitArray2, offset2, groupLength, eccInfo, generatorPolynom);
```
Comment (codebude) [inline QRCoder/QRCodeGenerator.cs:300]: Should this be?
```suggestion
                    var eccWordList = CalculateECCWords(bitArray, offset2, groupLength, eccInfo, generatorPolynom);
```

**Review by codebude — CHANGES_REQUESTED** [review]

File: QRCoder/QRCodeGenerator.cs (hunk 2 above)
Comment (Shane32) [reply QRCoder/QRCodeGenerator.cs]: Yeah, probably that was the only change needed.

Comment (Shane32) [reply QRCoder/QRCodeGenerator.cs]: I wasn't sure so I reverted the merge just so I could review my prior changes and reapply them.  It's all set now, I believe.  Tests pass locally anyway.

Comment (codebude) [reply QRCoder/QRCodeGenerator.cs]: Thanks. As soon as the action runner succeeds, I'll merge. :-)

**Review by codebude — APPROVED** [review]

**Pull request merged**